packages/cmr-client/src/EarthdataLogin.ts

```diff
@@ -1,202 +1,155 @@
-    let token = await this.retrieveEDLToken();
-    if (token === undefined) {
-      token = await this.createEDLToken();
-    }
-    return token;
+/**
+* The method for getting the Earthdata Login endpoint URL based on the EDL environment
+*
+* @returns {<string>} the endpoint URL
+*/
+const getEdlUrl = (env: string) : string => {
+  switch (env) {
+    case 'PROD':
+    case 'OPS':
+      return 'https://urs.earthdata.nasa.gov';
+    case 'UAT':
+      return 'https://uat.urs.earthdata.nasa.gov';
+    case 'SIT':
+    default:
+      return 'https://sit.urs.earthdata.nasa.gov';
   }
+};
 
-  /**
-   * This helper method is called in the functions that retrieve, create, and revoke EarthdataLogin
-   * tokens for error-handling. If API call made to the EarthdataLogin endpoint results in an error.
-   * The statuscode, statusmessage, error description, and error message are thrown and outputted.
-   *
-   * @throws {Error} - EarthdataLogin error
-   */
-  handleHttpError(error: Error) {
-    const statusCode = get(error, 'response.statusCode', error.code);
-    const statusMessage = get(error, 'response.statusMessage', error.message);
-    const responseErrorDescription = JSON.parse(get(error, 'response.body')).error_description;
-    const errorMessage = `EarthdataLogin error: ${responseErrorDescription},  statusCode: ${statusCode}, statusMessage: ${statusMessage}. Earthdata Login Request failed`;
+/**
+* This helper method is called in the functions that retrieve, create, and revoke EarthdataLogin
+* tokens for error-handling. If API call made to the EarthdataLogin endpoint results in an error.
+* The statuscode, statusmessage, error description, and error message are thrown and outputted.
+*
+* @returns {Error} - EarthdataLogin error
+*/
+const parseHttpError = (error: HTTPError, requestType: string): Error => {
+  const statusCode = error.response.statusCode;
+  const statusMessage = error.response.statusMessage || 'Unknown';
+  const errorBody = requestType === 'retrieve' || requestType === 'create' ? JSON.stringify(error.response.body) : error.response.body;
+  const message = `EarthdataLogin error: ${errorBody},  statusCode: ${statusCode}, statusMessage: ${statusMessage}. Earthdata Login Request failed`;
+  return new Error(message);
+};
 
-    throw new Error(errorMessage);
+/**
+  * The method for getting the token from the Earthdata Login endpoint. Sends a GET request
+  * with the users' base64 encoded username and password as a header for authorization. If the
+  * users' credentials are accepted the first unexpired token is retrieved, if one exists, and
+  * returned on their behalf, if not, an error is thrown. If the user does not have a token
+  * in Earthdata Login then undefined is returned to indicate to token needs to be created.
+  *
+  * @returns {string} the token
+*/
+export const retrieveEDLToken = async (username: string, password: string, edlEnv: string):
+Promise<string | undefined> => {
```

Comment:
It's a little easier to read if it's formatted like this:

```suggestion
export const retrieveEDLToken = async (
  username: string,
  password: string,
  edlEnv: string
): Promise<string | undefined> => {
```